protocol-units/bridge/shared/src/types.rs

```diff
@@ -135,13 +138,84 @@ impl From<Uint<256, 4>> for TimeLock {
 }
 
 #[derive(Deref, DerefMut, Debug, Clone, Copy, PartialEq, Eq)]
-pub struct Amount(pub u64);
+pub struct Amount(pub AssetType);
+/// The type of Asset being used
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+pub enum AssetType {
+	/// Where the first tuple value is `Eth` and the second tuple value is `Weth`  
+	EthAndWeth((u64, u64)),
```

Comment:
I know we've gone over this a few times, but I still don't understand why we can't have:
```rust
pub enum AssetType {
   Eth(u64)
   Weth(u64)
   EthAndWeth((u64, u64))
   Moveth(u64)
}
```

Otherwise we have code like 
`Amount(AssetType::EthAndWeth((1000,0))),` 

This reads weird because its not Eth and Weth, it's just Eth. Do you see what I mean?

Reply:
Yep, but then if we are working with the bridge service we have to handle unexpected types. Notice that we cannot use more than one type. Also, this streamlines eth and weth handling for any call to the eth side. What if we expect eth only but get weth too? That weth is lost in value. I just feel like and eth and weth variants is over engineering. We can call ethandweth something else though

Reply:
In rust any unexpected type at runtime would have to be `dyn` which its not here. These are all concrete types known at compile time. 

Any new `AssetType` could just be added as an enum variant, for eg `Btc`, `Sol`, `Usdc` etc. We can add as many as we want. So it becomes code as documentation. This would setup a nice pattern for configuration and extension. 

In your example, if we expect only eth but get weth too that would match on the `EthAndWeth` arm, and we handle that case.  I don't see the problem and see it only as a worthwhile improvement that improves readability, extensibility (more asset types) and semantics.   

Anything coming "out" from ETH initiator is always WETH.

Reply:
But then if we are handling BTC, SOL, USDC, etc, they would be new enum types, for sure, but they are not being handled as BTC and WBTC, they'll be just a single type. If we are handling BTC and WBTC from a chain that supports native and wrapped btc (Rootstock for example), We add a variant BtcAndWbtc((u64,u64)). This way it's easy to be explicit that this implementation, on this side of the bridge, handled two assets for the sake of compatibility with the chain.